### PR TITLE
document abstract adaptor methods

### PR DESCRIPTION
**Pull request by malmans2**

### CODE REVIEW

File: cads_adaptors/adaptors/__init__.py [hunk 1]
```diff
@@ -141,14 +141,53 @@ def apply_constraints(self, request: Request) -> dict[str, Any]:
...
+
+        Parameters
+        ----------
+        request : Request
+            Incoming request.
+
+        Returns
+        -------
+        dict[str, int]
+            Estimated costs.
```
Comment (EddyCMWF) [inline cads_adaptors/adaptors/__init__.py:155]:
```suggestion
            Estimated costs, where the key is the cost name/type (e.g. size/precise_size) and the value is the maximum cost for that type.
```

**Review by EddyCMWF — APPROVED** [review]
Approved if more detail is added to the estimate_costs return object

File: cads_adaptors/adaptors/__init__.py (hunk 1 above)
Comment (EddyCMWF) [reply cads_adaptors/adaptors/__init__.py:155]: something to this effect could be added to the description if you prefer. Essentially, other people need to know what the returned dictionary shuld contain

Comment (malmans2) [reply cads_adaptors/adaptors/__init__.py:155]: sounds good. I'll add the same wording in the confluence page

**Pull request merged**